### DISCUSSION TITLE
CAPSTONE ML - Image Classification - How to change validation keras to dataframe

### DISCUSSION THREAD

**Discussion by vetricko** [teamalgoritma/community#352] [Q&A] [answered]
Rencana mau predict data validation dengan model sedangkan setelah splitting data, data validasinya dalam bentuk keras 

<img width="679" alt="image" src="https://user-images.githubusercontent.com/65656955/192220821-cde7a206-f9fa-4128-8d10-6d8ed788eeda.png">

bagaimana cara mengubah data keras ke data frame?

Apakah ada cara lain untuk prepare data validasi sebelum di confusion matrix data train dan data validasi?

**Comment by fafilia** [comment, marked as answer]
Dear @vetricko ,

Mohon maaf pak disini kebutuhan untuk mengubah ke dataframe untuk apa ya? Karena untuk Keras sendiri memang mengharuskan ketika proses fitting(dengan data train dan validasi) bentuk datanya adalah berupa array. Hal ini nantinya juga digunakan untuk prediksi ke data test. Data test harus diperlakukan sama seperti data train dan data validasi. Setelah didapatkan hasil prediksi untuk data test, selanjutnya Bapak bisa melakukan decoding agar ketika dilakukan perhitungan confussion matrix datanya sama seperti label aktualnya. Contoh decoding sebagai berikut:

```
decode <- function(x){
  case_when(x == 0 ~ "beach",
            x == 1 ~ "forest",
            x == 2 ~ "mountain"
            )
}

pred_test <- sapply(pred_test, decode) 

```

**Reply by vetricko** [reply]
Terimakasih telah mencoba menjawab, sudah dapat codenya, ternyata ada di blog algoritma dari referensi capstone. Alasan mengubah dataframe untuk mengambil filename dan mengestract label kategori. dibawah adalah code yang sy dapat.

val_data <- data.frame(file_name = paste0("data_input/image_class/train/", val_image_array_gen$filenames)) %>% 
  mutate(class = str_extract(file_name, "beach|forest|mountain"))
